tests/capture/transforms/test_capture_single_qubit_fusion.py

```diff
@@ -0,0 +1,699 @@
+        assert jaxpr.eqns[2].primitive == qml.RY._primitive
+
+        inner_jaxpr = jaxpr.eqns[1].params["inner_jaxpr"]
+        assert len(inner_jaxpr.eqns) == 2
+        assert inner_jaxpr.eqns[0].primitive == qml.S._primitive
+        assert inner_jaxpr.eqns[1].primitive == qml.T._primitive
+
+    @pytest.mark.parametrize(
+        "selector, expected_ops",
+        [
+            (
+                0.2,
+                [
+                    qml.CNOT(wires=[0, 1]),
+                    qml.Hadamard(wires=[0]),  # H(0) and H(1) are not fused
+                    qml.Hadamard(wires=[1]),  # because they act on different wires
+                    qml.CNOT(wires=[0, 1]),
+                ],
+            ),
+            (
+                0.6,
+                [
+                    qml.CNOT(wires=[0, 1]),
+                    qml.Rot(np.pi / 2, 0.6, 0.0, wires=[0]),  # RX and S fused
+                    qml.CNOT(wires=[0, 1]),
+                ],
+            ),
+            (
+                0.8,
+                [
+                    qml.CNOT(wires=[0, 1]),
+                    qml.Rot(np.pi / 2, 0.8, 0.0, wires=[0]),  # RX and S fused
+                    qml.CNOT(wires=[0, 1]),
+                ],
+            ),
+        ],
+    )
+    def test_cond(self, selector, expected_ops):
+        """Test that operations inside a conditional block are correctly fused."""
+
+        @SingleQubitFusionInterpreter()
+        def circuit(x):
+
+            qml.CNOT(wires=[0, 1])
+
+            def true_branch(x):
+                qml.RX(x, wires=0)
+                qml.S(wires=0)
+
+            # pylint: disable=unused-argument
+            def false_branch(x):
+                qml.H(0)
+                qml.H(1)
+
+            qml.cond(x > 0.5, true_branch, false_branch)(x)
+            qml.CNOT(wires=[0, 1])
+
+        jaxpr = jax.make_jaxpr(circuit)(selector)
+
+        assert len(jaxpr.eqns) == 4
+        assert jaxpr.eqns[0].primitive == qml.CNOT._primitive
+        assert jaxpr.eqns[1].primitive == jax.lax.gt_p
+        assert jaxpr.eqns[2].primitive == cond_prim
+        assert jaxpr.eqns[3].primitive == qml.CNOT._primitive
+
+        collector = CollectOpsandMeas()
+        collector.eval(jaxpr.jaxpr, jaxpr.consts, selector)
+        jaxpr_ops = collector.state["ops"]
+
+        for op1, op2 in zip(jaxpr_ops, expected_ops):
+            # The qml.equal function does not recognize two qml.Rot operators
+            # as equivalent unless the input angles are exactly the same
+            assert op1.name == op2.name
+            assert qml.math.allclose(op1.parameters, op2.parameters)
+            assert op1.wires == op2.wires
+
+    def test_for_loop(self):
+        """Test that for operators inside a for loop are correctly fused."""
+
+        @SingleQubitFusionInterpreter()
+        def circuit(x):
+
+            qml.CNOT(wires=[0, 1])
+
+            @qml.for_loop(0, 1)
```

Comment:
If we increment this to one more iteration, this test nails down the cross-loop-boundary behaviour of `single_qubit_fusion` as well, which might be nice :) (Basically just like for `while` in the next test)